mozetl/hardware_report/summarize_json.py

```diff
@@ -0,0 +1,691 @@
+
+    if os.path.exists(file_name):
+        print "{} exists, we will overwrite it.".format(file_name)
+
+    # Our aggregated data is a JSON object.
+    json_entry = json.dumps(aggregated_data)
+
+    with open(file_name, "w") as json_file:
+        json_file.write("[" + json_entry.encode('utf8') + "]\n")
+
+
+def fetch_previous_state(s3_source_file_name, local_file_name):
+    """
+    This function fetches the previous state from S3's bucket and stores it locally.
+
+    Args:
+        s3_source_file_name: The name of the file on S3.
+        local_file_name: The name of the file to save to, locally.
+    """
+
+    # Fetch the previous state.
+    client = boto3.client('s3', 'us-west-2')
+    transfer = boto3.s3.transfer.S3Transfer(client)
+    key_path = S3_DATA_PATH + s3_source_file_name
+
+    try:
+        transfer.download_file(S3_PUBLIC_BUCKET, key_path, local_file_name)
+    except botocore.exceptions.ClientError as e:
+        # If the file wasn't there, that's ok. Otherwise, abort!
+        if e.response['Error']['Code'] != "404":
+            raise e
+        else:
+            print "Did not find an existing file at '{}'".format(key_path)
+
+
+def store_new_state(source_file_name, s3_dest_file_name):
+    """
+    Store the new state file to S3.
+
+    Args:
+        source_file_name: The name of the local source file.
+        s3_dest_file_name: The name of the destination file on S3.
+    """
+
+    client = boto3.client('s3', 'us-west-2')
+    transfer = boto3.s3.transfer.S3Transfer(client)
+
+    # Update the state in the analysis bucket.
+    key_path = S3_DATA_PATH + s3_dest_file_name
+    transfer.upload_file(source_file_name, S3_PUBLIC_BUCKET, key_path)
+
+
+def generate_report(start_date, end_date):
+    """ Generates the hardware survey dataset for the reference timeframe.
+
+    If the timeframe is longer than a week, split it in in weekly chunks
+    and process each chunk individually (eases backfilling).
+
+    The report for each week is saved in a local JSON file.
+
+    Args:
+        start_date: The date from which we start generating the report. If None,
+           the report starts from the beginning of the past week (Sunday).
+        end_date: The date the marks the end of the reporting period. This only
+           makes sense if a |start_date| was provided. If None, this defaults
+           to the end of the past week (Saturday).
+    """
+
+    # If no start_date was provided, generate a report for the past complete
+    # week.
+    last_week = moz_std.get_last_week_range()
+    date_range = (
+        moz_std.snap_to_beginning_of_week(
+            start_date, "Sunday") if start_date is not None else last_week[0],
+        end_date if (
+            end_date is not None and start_date is not None) else last_week[1]
+    )
+
+    # Connect to the longitudinal dataset.
+    sqlQuery = "SELECT " + "build," + "client_id," + "active_plugins," + "system_os," + "submission_date," + \
```

Comment:
Can you fix this SQL line to use a multiline string (`"""`) and look a bit more sql-y? I prefer:
```
SELECT
  col1,
  col2
FROM
  tablename
```

but whatever you'd like is fine.